cli/src/template_builder.rs

```diff
@@ -1063,6 +1064,58 @@ fn builtin_string_methods<'a, L: TemplateLanguage<'a> + ?Sized>()
             Ok(out_property.into_dyn_wrapped())
         },
     );
+    map.insert(
+        "replace",
+        |language, diagnostics, build_ctx, self_property, function| {
+            let ([pattern_node, replacement_node], [limit_node]) = function.expect_arguments()?;
+            let pattern = template_parser::expect_string_pattern(pattern_node)?;
+            let replacement_property =
+                expect_stringify_expression(language, diagnostics, build_ctx, replacement_node)?;
+
+            let is_regex = matches!(pattern, StringPattern::Regex(_) | StringPattern::RegexI(_));
+            let regex = pattern.to_regex();
+
+            let do_replace = move |haystack: &str, replace: &str, limit: Option<usize>| {
+                let haystack_bytes = haystack.as_bytes();
+                let replace_bytes = replace.as_bytes();
+                let result = match (is_regex, limit) {
+                    (true, None) => regex.replace_all(haystack_bytes, replace_bytes),
+                    (true, Some(n)) => regex.replacen(haystack_bytes, n, replace_bytes),
+                    (false, None) => {
+                        regex.replace_all(haystack_bytes, regex::bytes::NoExpand(replace_bytes))
+                    }
+                    (false, Some(n)) => {
+                        regex.replacen(haystack_bytes, n, regex::bytes::NoExpand(replace_bytes))
+                    }
+                };
+                std::str::from_utf8(&result).map(str::to_owned)
+            };
+
+            let out_property = if let Some(limit_node) = limit_node {
+                let limit_property =
+                    expect_isize_expression(language, diagnostics, build_ctx, limit_node)?;
```

Comment:
nit: `expect_usize_expression()` can be used.

BTW, `.replace(.., 0)` should replace nothing.

Reply:
It seems weird that negative limit underflows to "unlimited". It might be okay to clamp it to 0, though.

If you need an explicit option to specify "unlimited", it's probably better to add `replace(pattern, subst, count)` and `replace_all(pattern, subst)`.

Reply:
Does replace without a limit argument do a replace all? If so all is good and there shouldn't be a need for a separate function